sympy/polys/tests/test_densearith.py

```diff
@@ -679,6 +687,29 @@ def test_dmp_mul():
         [[K(2)], [K(1)]], [[K(3)], [K(4)]], 1, K) == [[K(5)], [K(4)]]
 
 
+def test_dup_series_mul():
+
+    # Check correctness against dup_mul with truncation
+    for i in range(50):
+        deg = randint(2, 200)
+        f = _dup_random(-20, 0, deg, ZZ)
+        g = _dup_random(0, 20, randint(1, 200), ZZ)
+
+        mul = dup_series_mul(f, g, deg, ZZ)
+        expected = dup_truncate(dup_mul(f, g, ZZ), deg, ZZ)
+        assert mul == expected
```

Comment:
Maybe let's replace the random tests here with deterministic tests that give full line coverage like
```
f = range(200)
g = [1] * 200

f = [n**2 for n in range(200)]
```
We want one or two small tests written explicitly but for large tests we can just compare dup_series_mul and dup_mul.